src/test/java/com/desiato/puresynth/models/UserControllerTest.java

```diff
@@ -0,0 +1,52 @@
+package com.desiato.puresynth.models;
+
+import com.desiato.puresynth.configurations.SecurityConfig;
+import com.desiato.puresynth.repositories.UserRepository;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Import(com.desiato.puresynth.configurations.SecurityConfig.class)
+public class UserControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @BeforeEach
+    @Transactional
+    public void setup() {
+        userRepository.deleteByEmail("test3@example.com");
+    }
+
+    @Test
+    @Transactional
```

Comment:
Tests shouldn't be `@Transactional`. It changes slightly the logic of how a transaction is managed. In those tests, we want to mimic the production environment as much as possible. 

Here, we're starting a bigger transaction context in the DB in reality. It could alter potential race conditions we could have in the app, and we want our tests to catch real issues.